src/Http/Shared/CookieHeaderParserShared.cs

```diff
@@ -107,6 +114,13 @@ private static int GetNextNonEmptyOrWhitespaceIndex(StringSegment input, int sta
         // If we have a separator, skip the separator and all following whitespaces. If we support
         // empty values, continue until the current character is neither a separator nor a whitespace.
         separatorFound = true;
+
+        //We accept single separator, but separator at the end of a value is ignored.
+        if (input.Length > 1 && current + 1 == input.Length)
+        {
+            return current;
+        }
+
```

Comment:
The tests now pass even if I remove this new segment. The other fix above is sufficient.
```suggestion
```

Reply:
Ah yes, I had it already locally but forget to commit then. Now the suggestions are applied, let's see the pipelines.